table/tables/partition_test.go

```diff
@@ -2349,3 +2356,653 @@ func TestPruneModeWarningInfo(t *testing.T) {
+		pk := getNewPK(pkMap, "-o")
+		pkArray = append(pkArray, pk)
+		values := getValues(pk, false)
+		tk.MustExec(`insert into t values ` + values)
+		tk.MustExec(`insert into t2 values ` + values)
+	}
+	// TODO: Also test without analyze table
+	tk.MustExec(`analyze table t`)
+	tk.MustExec(`analyze table t2`)
+	tk.MustQuery(`select * from t except select * from t2`).Check(testkit.Rows())
+	tk.MustQuery(`select * from t2 except select * from t`).Check(testkit.Rows())
+
+	// How to arrange data for possible collisions?
+	// change both PK column, SK column and non indexed column!
+	// Run various changes in transactions, in two concurrent sessions
+	// + mirror those transactions on a copy of the same table and data without DDL
+	// to verify expected transaction conflicts!
+	// We should try to collide:
+	// Current data : 1-1000
+	// insert vN    1-200 // random order, random length of transaction?
+	// insert vN-1 100-300 // interleaved with vN, random order+length of txn?
+	// update vN    1-20, 100-120, 200-220, 300-320..
+	// update vN-1  10-30, 110-130, 210-230, ...
+	// delete vN
+	// delete vN-1
+	//               insert      update       delete <-
+	//  insert
+	//  update
+	//  delete
+	// TODO: Also make sure we update the PK so it moves between different before and after partitions
+	// TODO: Also check if non-affected indexes in UPDATE is touched?
+	tk2 := testkit.NewTestKit(t, store)
+	tk2.MustExec("use " + schemaName)
+	tk2.MustQuery(`select count(*) from (select a from t except select a from t2) a`).Check(testkit.Rows("0"))
+	tk2.MustQuery(`select count(*) from (select a from t2 except select a from t) a`).Check(testkit.Rows("0"))
+	currentState := model.StateNone
+	transitions := 0
+	var currTbl table.Table
+	currSchema := sessiontxn.GetTxnManager(tk2.Session()).GetTxnInfoSchema()
+	prevTbl, err := currSchema.TableByName(model.NewCIStr(schemaName), model.NewCIStr("t"))
+	require.NoError(t, err)
+	var hookErr error
+	hook.OnJobRunBeforeExported = func(job *model.Job) {
+		if hookErr != nil {
+			// Enough to find a single error
+			return
+		}
+		if job.Type == model.ActionReorganizePartition && job.SchemaState != currentState {
+			transitions++
+			// use random generation to possibly trigger txn collisions / deadlocks?
+			// insert (dup in new/old , non dup)
+			// update (dup in new/old , non dup as in same old/new partition -> update, different new/old -> insert + delete)
+			// delete
+			// verify with select after commit?
+
+			logutil.BgLogger().Info("State before ins/upd/del", zap.Int("transitions", transitions),
+				zap.Int("rows", len(pkMap)), zap.Stringer("SchemaState", job.SchemaState))
+			if job.SchemaState == model.StateDeleteReorganization {
+				//tk2.MustQuery(`explain format='brief' select count(distinct a) from t`).Check(testkit.Rows())
+			}
+			//tk2.MustQuery(`(select a,c from t except select a,c from t2) union (select a,0 from t except select a,0 from t2)`).Check(testkit.Rows())
```

Comment:
Should we remove the commented codes?

Reply:
Ah, yes :) (there were left over after debugging...)

Reply:
Fixed